src/routes/InstancesPage/InstancesToolbarSearchFilter.js

```diff
@@ -0,0 +1,230 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ToolbarToggleGroup,
+  ToolbarGroup,
+  ToolbarFilter,
+  SelectVariant,
+  SelectOption,
+  Select,
+  ToolbarItem,
+  InputGroup,
+  TextInput,
+  Button,
+} from '@patternfly/react-core';
+import { FilterIcon, SearchIcon } from '@patternfly/react-icons';
+import SelectSingle from '../../components/SelectSingle';
+
+function InstancesToolbarSearchFilter({ filters, setFilters }) {
+  const [selectedFilter, setSelectedFilter] = useState('Name');
+  // local state for input values
+  const [inputName, setInputName] = useState('');
+  const [inputOwner, setInputOwner] = useState('');
+  // local state for Select isExpanded values
+  // @TODO: We can refactor the SelectSingle component to be more reusable for the usecase in this component as well. Then we don't need to keep this state here.
+  const [isRegionExpanded, setIsRegionExpanded] = useState(false);
+  const [isStatusExpanded, setIsStatusExpanded] = useState(false);
+
+  // TODO: Extract into separate utils file to be reused in other cases
+  function onDeleteChip(type = '', id = '') {
+    if (type) {
+      setFilters((prevFilters) => {
+        const newFilters = { ...prevFilters };
+        newFilters[type.toLowerCase()] = newFilters[type.toLowerCase()].filter(
+          (s) => s !== id
+        );
+        return newFilters;
+      });
+    } else {
+      setFilters({
+        region: [],
+      });
+    }
+  }
+
+  // TODO: Extract into separate utils file to be reused in other cases
+  function onDeleteChipGroup(type) {
+    setFilters((prevFilters) => {
+      const newFilters = { ...prevFilters };
+      newFilters[type.toLowerCase()] = [];
```

Comment:
It would be cool to have the `filters` type only accept lowercase strings when this is converted to TypeScript, so that someone can't accidentally forget to call the `toLowerCase` method in utility functions.

Reply:
Yea that's a good point. We can keep it lowercase then. I'll try to make it more consistent

Reply:
Oh actually, I realized we can't do what you suggested. When we render the chips, we want to have the category be capitalized (ie. `Name`), and that's passed through the `categoryName` prop. I believe that's why we need to lowercase here. I took this from the PF Doc example. I'm guessing that's why they did it like that. If we standardize to be lowercase, then the rendered category will also be lowercase.

Reply:
It's odd they do it like that. It would have been nice to have a distinction between the rendered category name and the value for the filter. What if we wanted a filter name with an underscore or something. I guess they would suggest not doing that and then changing it wherever that modified field is needed

Reply:
The hacky PF is fine for first pass, but I think we should move in the direction of separating _values_ from _display_. A thing's value should not be dependent on whether it is displayed uppercase or Titlecase or whatever.

One solution is have the filters be an array of objects like this:
```
[
  {
    value: 'ready',
    text: 'Ready',
  },
  {
    value: 'not_ready',
    text: 'Not ready',
  },
]
```

Reply:
Ah, I see. I was looking at it from the point of view of the component, but that makes sense. I can see if we can structure the filters state object differently

Reply:
I guess there is a way (I think)
```
export interface ToolbarFilterProps extends ToolbarItemProps {
  /** An array of strings to be displayed as chips in the expandable content */
  chips?: (string | ToolbarChip)[];
  /** Callback passed by consumer used to close the entire chip group */
  deleteChipGroup?: (category: string | ToolbarChipGroup) => void;
  /** Callback passed by consumer used to delete a chip from the chips[] */
  deleteChip?: (category: string | ToolbarChipGroup, chip: ToolbarChip | string) => void;
  /** Customizable "Show Less" text string for the chip group */
  chipGroupExpandedText?: string;
  /** Customizeable template string for the chip group. Use variable "${remaining}" for the overflow chip count. */
  chipGroupCollapsedText?: string;
  /** Content to be rendered inside the data toolbar item associated with the chip group */
  children: React.ReactNode;
  /** Unique category name to be used as a label for the chip group */
  categoryName: string | ToolbarChipGroup;
  /** Flag to show the toolbar item */
  showToolbarItem?: boolean;
}
```
Looks like `categoryName` can be a `ToolbarChipGroup` type that is 
```
export interface ToolbarChipGroup {
  /** A unique key to identify this chip group category */
  key: string;
  /** The category name to display for the chip group */
  name: string;
}
```

Reply:
I'll definitely try to add Typescript after this lol